core/src/rpc_pubsub.rs

```diff
@@ -570,6 +570,8 @@ mod tests {
             .process_transaction(&tx)
             .unwrap();
         rpc.subscriptions.notify_subscribers(0, &bank_forks);
+        // allow 200ms for notification thread to wake
+        std::thread::sleep(Duration::from_millis(200));
```

Comment:
Why do we need these sleeps?   If we depend on them, we're looking at hard to reproduce CI failures down the road

Reply:
* enqueue is on a separate thread from dequeue/process
* we're using `try_recv()` on the channel instead of blocking `recv()`, with an empty iteration delay of 100ms
* this gives us a chance every 100ms to wake up and process (or detect the exit criterion)
* given the structure of non-blocking receive to allow cancellation, it wasn't immediately clear how to trigger "instantly"
* let me know if you have an idea!

Reply:
Run `receiver.poll()` in a loop?